client/reader/full-post/controller.js

```diff
@@ -61,6 +93,19 @@ export function feedPost( context, next ) {
 			referralStream={ context.lastRoute }
 		/>
 	);
+
+	if ( isUserLoggedIn( state ) ) {
+		context.secondary = (
+			<AsyncLoad
+				require="calypso/reader/sidebar"
+				path={ context.path }
+				placeholder={ null }
+				returnPath={ lastRoute }
+				onClose={ closer }
```

Comment:
While we are already handling the return path and back functionality with the onClose, supplying the returnPath allows the breadcrumb url for the anchor link to be accurate.